server/app/mutations/grid_services/update.rb

```diff
@@ -120,7 +120,7 @@ def build_grid_service_envs(env)
       new_env = GridService.new(env: env).env_hash
       current_env = self.grid_service.env_hash
       new_env.each do |k, v|
-        if v.empty? && current_env[k]
+        if (v.nil? || v.empty?) && current_env[k]
```

Comment:
Implemented the service env validation over in  #2058 (https://github.com/kontena/kontena/pull/2058/commits/2230cfb5e3fa814c8e20d9362b058aeeaea0be81).

Trying to do the same validation here would result in HTTP 500 errors as the `handle_service_outcome_errors`  would be unable to handle the resulting `Mutations::ErrorArray`.